cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamService.java

```diff
@@ -97,4 +112,39 @@ protected ScheduledExecutorService executor() {
     executor = Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("heartbeats-scheduler"));
     return executor;
   }
+
+  /**
+   * Get the size of events ingested by a stream since its creation, in bytes.
+   * @param streamId id of the stream
+   * @return Size of events ingested by a stream since its creation
+   * @throws IOException when getting an error retrieving the metric
+   */
+  protected long getStreamEventsSize(Id.Stream streamId) throws IOException {
+    MetricDataQuery metricDataQuery = new MetricDataQuery(
```

Comment:
ok got it. the metric is a counter over all time that can expire if nobody writes to the stream after TTL amount of time, but it's not a running window or anything like that.  all good then.